plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/SWTBrowserRegistry.java

```diff
@@ -0,0 +1,104 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.registry;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.ModelPreferences;
+import org.jkiss.dbeaver.model.preferences.DBPPreferenceStore;
+import org.jkiss.dbeaver.utils.RuntimeUtils;
+import org.jkiss.utils.CommonUtils;
+
+import java.util.Arrays;
+
+
+public class SWTBrowserRegistry {
+
+    private static final String INTERNAL_BROWSER_PROPERTY = "org.eclipse.swt.browser.DefaultType";
+
+    //Windows versions which could not have Edge as an available browser, or need to install it manually
+    private static final String[] LEGACY_OS = {
+        "Windows 98", //$NON-NLS-1$
+        "Windows XP", //$NON-NLS-1$
+        "Windows Vista", //$NON-NLS-1$
+        "Windows 7", //$NON-NLS-1$
+        "Windows 8", //$NON-NLS-1$
+        "Windows 8.1", //$NON-NLS-1$
+    };
+
+    public enum BrowserSelection {
+        EDGE("Microsoft Edge"),
+        IE("Internet Explorer");
+
+        private final String name;
+
+        BrowserSelection(@NotNull String name) {
+            this.name = name;
+        }
+
+        public String getFullName() {
+            return name;
+        }
+    }
+
+    /**
+     * @return Get selected browser or default browser if none is selected
+     */
+    public static BrowserSelection getActiveBrowser() {
+        DBPPreferenceStore preferences = ModelPreferences.getPreferences();
+        String type = preferences.getString(ModelPreferences.CLIENT_BROWSER);
+        if (CommonUtils.isEmpty(type)) {
+            return getDefaultBrowser();
+        } else {
+            return CommonUtils.valueOf(BrowserSelection.class, type.toUpperCase());
+        }
+    }
+
+    /**
+     * @return returns browser by default, depends on OS
+     */
+    public static BrowserSelection getDefaultBrowser() {
+        if (RuntimeUtils.isWindows() && Arrays.stream(LEGACY_OS)
+            .anyMatch(it -> it.equalsIgnoreCase(System.getProperty("os"
+                + ".name")))) {
+            return BrowserSelection.IE;
+        } else {
+            return BrowserSelection.EDGE;
+        }
+    }
+
+    /**
+     * Overrides default browser, we want to use Edge for newer version
+     * if the user didn't specify otherwise
+     */
+    public static void overrideBrowser() {
+        if (getActiveBrowser() != null) {
+            System.setProperty(INTERNAL_BROWSER_PROPERTY, getActiveBrowser().name().toLowerCase());
+        }
+    }
+
+    /**
+     * By passing down BrowserSelection sets browser to be used by eclipse
+     *
+     * @param browser selected browser
+     */
+    public static void setActiveBrowser(BrowserSelection browser) {
+        DBPPreferenceStore preferences = ModelPreferences.getPreferences();
+        preferences.setValue(ModelPreferences.CLIENT_BROWSER, browser.name().toLowerCase());
```

Comment:
Why `toLowerCase()`?

Reply:
because the name() returns upper cased constant if we pass it down to eclipse I am not sure it will work. So we set it to lower case

Reply:
It will work. No uncertainty, please